core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -380,6 +381,21 @@ public void removeBlocks(List<Long> blockIds, boolean delete) throws Unavailable
     }
   }
 
+  @Override
+  public void validateBlocks(Function<Long, Boolean> validator) throws UnavailableException {
+    List<Long> invalidBlocks = new ArrayList<>();
+    for (long blockId : mBlocks.keySet()) {
+      if (!validator.apply(blockId)) {
+        LOG.debug("Block {} has no corresponding file metadata, marked as invalid.", blockId);
+        invalidBlocks.add(blockId);
+      }
+    }
+    if (!invalidBlocks.isEmpty()) {
+      LOG.warn("Deleting {} invalid blocks from the Block Master.", invalidBlocks.size());
```

Comment:
list the block ids? Production usually doesn't run with `debug`, and this might be hard to repro

Reply:
I've changed the logging a bit